### PR TITLE
Add image server option, /manga/ intent to mangadex

### PR DESCRIPTION
**Pull request by ZaneHannanAU**
Adds intents for `/manga/`.
Adds an option to override image server parameter.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Editor configuration, see https://editorconfig.org
```
Comment (arkon) [inline .editorconfig]: Remove this file.

File: src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt [hunk 2]
```diff
@@ -532,13 +537,28 @@ open class Mangadex(override val lang: String, private val internalLang: String,
...
+            key = SERVER_PREF_Title
+            title = SERVER_PREF_Title
+            entries = arrayOf("Auto", "North America", "North America 2", "Europe", "Europe 2", "Rest of the World")
+            entryValues = arrayOf("0", "na", "na2", "eu", "eu2", "row")
+            summary = "%s"
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = this.findIndexOfValue(selected)
+                preferences.edit().putInt(SERVER_PREF, index).commit()
```
Comment (arkon) [inline src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt]: Shouldn't this be `putString`?

Comment (ZaneHannanAU) [reply src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt]: It's an index, so no.

File: .editorconfig (hunk 1 above)
Comment (ZaneHannanAU) [reply .editorconfig]: Ah, oops.

File: src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt (hunk 2 above)
Comment (ZaneHannanAU) [reply src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt]: Or at least I don't think so.

Comment (ZaneHannanAU) [reply src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt]: I'll try it.

Comment (ZaneHannanAU) [reply src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt]: … ah it makes more sense now. Sorry thank you.

**Comment by arkon** [comment]
Did you test this?

**Comment by ZaneHannanAU** [comment]
@arkon no I've been writing all the edits off of my phone... I haven't had a dev environment and android studio makes zero sense to me…
I mean I presume I can follow the .travis file but I'd need to run from my laptop of desktop for that

**Comment by ZaneHannanAU** [comment]
Or at least to get everything in order to set it up

**Comment by CarlosEsco** [comment]
travis will verify it builds, but that it works

**Comment by ZaneHannanAU** [comment]
@CarlosEsco I understand that, but I can't currently run android studio off of my phone. Or at least, I'm 90% sure that I can't… it might be possible to but I'll need to spend some time with it.

**Comment by CarlosEsco** [comment]
Just saying that this won't be merged until tested

**Comment by ZaneHannanAU** [comment]
I just need to set up everything to test it, so that's understandable

**Comment by ZaneHannanAU** [comment]
Okay, test succeeded.

queue

sync

options

open

File: src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt [hunk 3]
```diff
@@ -532,13 +537,29 @@ open class Mangadex(override val lang: String, private val internalLang: String,
...
+        }
 
         screen.addPreference(myPref)
         screen.addPreference(thumbsPref)
+        screen.addPreference(serverPref)
     }
 
     private fun getShowR18(): Int = preferences.getInt(SHOW_R18_PREF, 0)
     private fun getShowThumbnail(): Int = preferences.getInt(SHOW_THUMBNAIL_PREF, 0)
+    private fun getServer(): String = preferences.getString(SERVER_PREF, "")
```
Comment (arkon) [inline src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt]: Should the default be `"0"`?

File: src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt [hunk 4]
```diff
@@ -532,13 +537,29 @@ open class Mangadex(override val lang: String, private val internalLang: String,
...
+            key = SERVER_PREF_Title
+            title = SERVER_PREF_Title
+            entries = arrayOf("Auto", "North America", "North America 2", "Europe", "Europe 2", "Rest of the World")
+            entryValues = arrayOf("0", "na", "na2", "eu", "eu2", "row")
+            summary = "%s"
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = this.findIndexOfValue(selected)
+		val entry = entryValues.get(index) as String
```
Comment (arkon) [inline src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt]: Nitpick: indentation

File: src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt (hunk 3 above)
Comment (ZaneHannanAU) [reply src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/Mangadex.kt]: ... probably.

**Pull request merged**